Library/Homebrew/dev-cmd/bump.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  module_function
+
+  def bump_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `bump` [<options>]
+
+        Display out-of-date brew formulae and the latest version available.
+        Also displays whether a pull request has been opened with the URL.
+      EOS
+      flag "--formula=",
+           description: "Return results for package by name."
+      flag "--limit=",
+           description: "Limit number of package results returned."
+      switch :verbose
+      switch :debug
+    end
+  end
+
+  def bump
+    bump_args.parse
+
+    requested_formula = Homebrew.args.formula
+    requested_formula&.downcase!
+
+    if requested_formula && !get_formula_details(requested_formula)
+      ohai "Requested formula #{requested_formula} is not valid Homebrew formula."
+      return
+    end
+
+    outdated_repology_packages = if requested_formula
+      Repology.single_package_query(requested_formula)
+    else
+      Repology.parse_api_response
+    end
+
+    if requested_formula && outdated_repology_packages.nil?
+      ohai "The requested formula, #{requested_formula}, is up-to-date."
+      puts "Current version: #{get_formula_details(requested_formula).version}"
+      return
+    end
+
+    outdated_packages = validate_and_format_packages(outdated_repology_packages)
+    display(outdated_packages)
+  end
+
+  def validate_and_format_packages(outdated_repology_packages)
```

Comment:
Might be worth moving some of this method (and those below) into a module (either one of the existing ones or a new one) so they can be more easily tested.